src/test/java/edu/cornell/kfs/concur/batch/service/impl/MockConcurRequestV4Backend.java

```diff
@@ -0,0 +1,204 @@
+                .calculateSearchStartIndexForLastPageOfResults(totalCount, limit);
+        
+        Stream.Builder<ConcurRequestV4OperationDTO> operationsBuilder = Stream.builder();
+        if (startIndex - limit >= 0) {
+            operationsBuilder.add(buildOperation(queryParameters, ConcurApiOperations.PREV, startIndex - limit));
+        }
+        if (startIndex + limit < totalCount) {
+            operationsBuilder.add(buildOperation(queryParameters, ConcurApiOperations.NEXT, startIndex + limit));
+        }
+        operationsBuilder.add(buildOperation(queryParameters, ConcurApiOperations.FIRST, 0));
+        operationsBuilder.add(buildOperation(queryParameters, ConcurApiOperations.LAST, startIndexForLastPage));
+        List<ConcurRequestV4OperationDTO> operations = operationsBuilder.build()
+                .collect(Collectors.toUnmodifiableList());
+        
+        ConcurRequestV4ListingDTO requestListing = new ConcurRequestV4ListingDTO();
+        requestListing.setTotalCount(Integer.valueOf(totalCount));
+        requestListing.setListItems(unboundedResults.subList(startIndex, endIndex));
+        requestListing.setOperations(operations);
+        return requestListing;
+    }
+
+    private ConcurRequestV4OperationDTO buildOperation(Map<String, String> queryParameters,
+            String name, int startIndex) {
+        Map<String, String> operationParameters = new HashMap<>(queryParameters);
+        operationParameters.put(ConcurApiParameters.START, String.valueOf(startIndex));
+        
+        ConcurRequestV4OperationDTO operation = new ConcurRequestV4OperationDTO();
+        operation.setName(name);
+        operation.setHref(UrlFactory.parameterizeUrl(baseRequestV4Url, operationParameters));
+        return operation;
+    }
+
+    private Date getExistingDateParameter(Map<String, String> queryParameters, String key) {
+        String value = getExistingParameter(queryParameters, key);
+        try {
+            return ConcurUtils.parseUTCDateToDateTime(value).toDate();
+        } catch (RuntimeException e) {
+            throw new IllegalArgumentException("Query has badly formatted date parameter " + key, e);
+        }
+    }
+
+    private int getExistingIntParameter(Map<String, String> queryParameters, String key) {
+        String value = getExistingParameter(queryParameters, key);
+        try {
+            return Integer.parseInt(value);
+        } catch (NumberFormatException e) {
+            throw new IllegalArgumentException("Query has badly formatted int parameter " + key, e);
+        }
+    }
+
+    private String getExistingParameter(Map<String, String> queryParameters, String key) {
+        String value = queryParameters.get(key);
+        if (StringUtils.isBlank(value)) {
+            throw new IllegalArgumentException("Query has blank or missing parameter: " + key);
+        }
+        return value;
+    }
+
+    private static final class RequestEntry {
```

Comment:
what is the purpose of this class, I am not understanding how it is used.

Reply:
This class is meant to encapsulate the travel requests that are present on the mock backend server. Each instance contains three objects for a specific travel request: Its fixture, its list item DTO, and its report/detail DTO. It was meant to avoid reconstructing the DTOs on each query call, and to lay the groundwork for testing updates to the DTOs when a future user story implements the travel request workflow handling.

Upon further consideration, though, I might be able to simplify this by setting up some lazy DTO construction and removing this helper class altogether. I'll look into cleaning this up if possible.